### ISSUE TITLE
Types for NavigationControl(minor)

### ISSUE THREAD

**Issue by guimochila** [maplibre/maplibre-gl-js#1754]
**maplibre-gl-js version**: 2.4.0

**browser**: Chrome Version 106.0.5249.119 

### Steps to Trigger Behavior

I can see that in the [documentation example](https://maplibre.org/maplibre-gl-js-docs/api/markers/#navigationcontrol) when creating a `new NavigationControl()` the NavigationOptions are optional:

![Screenshot 2022-10-19 at 15 58 19](https://user-images.githubusercontent.com/4387043/196712179-3eb9deb1-d598-44ce-b69b-8c6f0d90ebf7.png)

However, in the types NagivationOptions are required and its properties are optional:

```ts
export declare type NavigationOptions = {
	showCompass?: boolean;
	showZoom?: boolean;
	visualizePitch?: boolean;
};
export declare class NavigationControl implements IControl {
	_map: Map;
	options: NavigationOptions;
	_container: HTMLElement;
	_zoomInButton: HTMLButtonElement;
	_zoomOutButton: HTMLButtonElement;
	_compass: HTMLButtonElement;
	_compassIcon: HTMLElement;
	_handler: MouseRotateWrapper;
	constructor(options: NavigationOptions);
	_updateZoomButtons(): void;
	_rotateCompassArrow(): void;
	onAdd(map: Map): HTMLElement;
	onRemove(): void;
	_createButton(className: string, fn: (e?: any) => unknown): HTMLButtonElement;
	_setButtonTitle(button: HTMLButtonElement, title: string): void;
}
```

So when creating a `new NavigationControl()` in my code I get a typescript error if I do like the example:

![Screenshot 2022-10-19 at 16 01 12](https://user-images.githubusercontent.com/4387043/196712878-f309364f-465c-4efc-ae8e-febbd1bac15c.png)

This is fixed by passing an empty object to `new NavigationControl({})`.

This is something that should be updated in the examples page or the options for NavigationControl should be optional?

**Comment by HarelM** [comment]
Looking at the `NavigationControl` constructor, it seems the code doesn't assume the options are provided as can be seen here:
https://github.com/maplibre/maplibre-gl-js/blob/55ab86c88fd6b026bbc3d2d6d7502f29f901cd2c/src/ui/control/navigation_control.ts#L47
Changing the signature of this constructor will solve this issue I guess.
Feel free to submit a PR :-)

**Comment by HarelM** [comment]
Fixed by #1756
